src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -130,12 +139,7 @@ public override bool Equals([NotNullWhen(true)] object? obj)
                 // fall through method handle check
```

Comment:
Could any of the checks above be removed now that the "slow" path is not that slow?